src/Compilers/CSharp/Portable/Symbols/NullableAnnotationExtensions.cs

```diff
@@ -108,13 +127,18 @@ internal static ImmutableArray<ITypeSymbol> GetPublicSymbols(this ImmutableArray
         internal static CodeAnalysis.NullableAnnotation ToPublicAnnotation(TypeSymbol type, NullableAnnotation annotation) =>
             annotation switch
             {
-                CSharp.NullableAnnotation.Annotated => CodeAnalysis.NullableAnnotation.Annotated,
-                CSharp.NullableAnnotation.NotAnnotated => CodeAnalysis.NullableAnnotation.NotAnnotated,
+                NullableAnnotation.Annotated => CodeAnalysis.NullableAnnotation.Annotated,
+                NullableAnnotation.NotAnnotated => CodeAnalysis.NullableAnnotation.NotAnnotated,
+
                 // A value type may be oblivious or not annotated depending on whether the type reference
                 // is from source or metadata. (Binding using the #nullable context only when setting the annotation
                 // to avoid checking IsValueType early.) The annotation is normalized here in the public API.
-                CSharp.NullableAnnotation.Oblivious when type.IsValueType => CodeAnalysis.NullableAnnotation.NotAnnotated,
-                CSharp.NullableAnnotation.Oblivious => CodeAnalysis.NullableAnnotation.None,
+                NullableAnnotation.Oblivious when type.IsValueType => CodeAnalysis.NullableAnnotation.NotAnnotated,
+                NullableAnnotation.Oblivious => CodeAnalysis.NullableAnnotation.None,
+
+                // May occur as type argument in symbols in diagnostic
```

Comment:
>// May occur as type argument in symbols in diagnostic [](start = 16, length = 54)

I think we should be able to avoid this if we use ConstructedFrom for diagnostic instead.

Reply:
I think we should really make an effort to not leak a symbol with an annotation like that. Otherwise, equality is likely to misbehave, etc. 

---
In reply to: [536200456](https://github.com/dotnet/roslyn/pull/49723#discussion_r536200456) [](ancestors = 536200456)